### ISSUE TITLE
[gatsby-plugin-preload-fonts] handle trailing slashes in onRenderBody

### ISSUE THREAD

**Issue by nibtime** [gatsbyjs/gatsby#21814]
## Description

I used the plugin and observed, that in some HTML files the preload links for the fonts were missing.
Then I saw, that in `font-preload-cache.json`, some routes contain a trailing slash, but I generate all my pages without a trailing slash. For those routes, the lookup in the asset map fails.

https://github.com/gatsbyjs/gatsby/blob/f585eb096cc2f51189d50d886f1c11547a83d2fb/packages/gatsby-plugin-preload-fonts/src/gatsby-ssr.js#L22

This could be replaced with

```js
const assetsOfPath = cache.assets[pathname] ? cache.assets[pathname] : cache.assets[pathname + "/"]
if (!assetsOfPath) {
  return
}
const assets = Object.keys(assetsOfPath)
```

### Steps to reproduce
Generate some pages at paths with no trailing slashes. I am not sure why sometimes paths are written with trailing slashes and sometimes not in `font-preload-cache.json`. However, if that happens, lookups will fail in `onRenderBody`.

### Expected result
Preload links should be written in all HTML files with a path match , independent of trailing slashes

### Actual result
Preload links are not written in HTML files with a non-trailing-slash path, if asset map contains the same path with a trailing slash as a key for the assets

**Comment by pvdz** [comment]
Thanks for reporting this :) Do you think this is related to #19543?

**Comment by nibtime** [comment]
Hi @pvdz 
I looked into #19543. Could be related, however there it is reported, that the redirect to a trailing slash only happens in production mode, but the `gatsby-preload-fonts` runs against a development build. It uses a GraphQL query to fetch all routes.

https://github.com/gatsbyjs/gatsby/blob/22ead1932af0d5e2a58e3e524fe0bcd2c1ae63ec/packages/gatsby-plugin-preload-fonts/src/prepare/fetch-routes.js#L8-L14

I checked the query results in GraphiQL, all routes are returned without a trailing slash as intended.

However, when navigating to the page in the browser, trailing slashes are added for most pages (not for all pages though, which is strange, since the pages are created with the same logic in `createPages`.). I do create redirects, however they are not targeted by `gatsby-preload-fonts`, as it iterates over the results of above GraphQL query, which all are fully qualified page paths.

It think the problem is here:

https://github.com/gatsbyjs/gatsby/blob/22ead1932af0d5e2a58e3e524fe0bcd2c1ae63ec/packages/gatsby-plugin-preload-fonts/src/prepare/index.js#L57-L69

In this callback, the path is obtained from the Puppeteer page object, which has the same URL as navigating to the page with Chrome and thus has a trailing slash. Then the path with the trailing slash is used as the key for the assets.

The pathname that is passed to `onRenderBody` in `gatsby-ssr.js` however does not have a trailing slash and that's why the lookup for that pathname fails (it obviously would just work fine if the page was created with a trailing slash).

https://github.com/gatsbyjs/gatsby/blob/399412af29131f17d55934e13448633b63a8a169/packages/gatsby-plugin-preload-fonts/src/gatsby-ssr.js#L16-L22

But all the trouble can be avoided by simply make a fallback lookup with a trailing slash. I will send a PR.
